src/kudu/consensus/metadata.proto

```diff
@@ -101,6 +104,8 @@ message CommitRulePredicatePB {
 message CommitRulePB {
   required QuorumMode mode = 1;
   repeated CommitRulePredicatePB rule_predicates = 2;
+  // Use quorum_id instead of region for flexiraft quorum
+  optional bool use_quorum_id = 3;
```

Comment:
What would be the advantages/disadvantages of setting this as another quorum mode? Does that make things hard to set elsewhere?

Right now Quorum Mode is SINGLE_REGION_DYNAMIC, which to me would have been confusing.

Reply:
Yeah, SINGLE_REGION_DYNAMIC becomes confusing, as it really means "SINGLE_QUORUM_DYNAMIC".

But in general I feel like "use_quorum_id" is different than (orthogonal to) commit mode. Mode defines what quorum requirement for commit and leader election, and use_quorum_id defines how we define a quorum.

For example, in the future, we could have both mode = STATIC_QUORUM but use_quorum_id can be true/false. For example,  it means is that it requires majority of two quorums, but the quorum can be a region/sub-region.

Reply:
I also thought about enum. I think enum indeed makes more sense. My reason is that when use_quorum_id = False, it is not clear what is the quorum definition. 

How is this?
  enum QuorumType {
    REGION = 0;
    QUORUM_ID = 1;
  };